pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -142,8 +142,16 @@ def __init__(
             B. Strategy > Accelerator/precision/plugins
             C. TODO When multiple flag set to the same thing
         """
+        # Raise a warning when benchmark and deterministic are True
+        if benchmark and deterministic:
+            rank_zero_warn(
+                "You passed `deterministic=True` and `benchmark=True`. Note that PyTorch ignores"
+                " torch.backends.cudnn.deterministic=True when torch.backends.cudnn.benchmark=True.",
+                category=RuntimeWarning,
```

Comment:
Why override the default `UserWarning`? This will only be shown after an user action, meaning, setting both of these flags